### DISCUSSION TITLE
macOS Setup with mpv

### DISCUSSION THREAD

**Discussion by b1n4ry5010** [jeffvli/feishin#139] [Ideas]
Hi, not really an 'idea' but just a tip when it comes to using Feishin on macOS.

The web player works out of the box, but in order to use mpv you will have to do the following:

1. Remove the .app extension from the mpv application, turning it into a browsable folder
2. In Settings > Playback, set MPV Executable Path to the mpv Unix Executable (Contents > MacOS > mpv)

This wasn't immediately clear to me, so maybe should be included somewhere in the readme?

**Comment by davin900** [comment]
> Hi, not really an 'idea' but just a tip when it comes to using Feishin on macOS.
> 
> The web player works out of the box, but in order to use mpv you will have to do the following:
> 
> 1. Remove the .app extension from the mpv application, turning it into a browsable folder
> 2. In Settings > Playback, set MPV Executable Path to the mpv Unix Executable (Contents > MacOS > mpv)
> 
> This wasn't immediately clear to me, so maybe should be included somewhere in the readme?

Wow thank you this helped a lot! I still had to figure out how to get the mpv files out of quarantine too.

**Comment by owldown** [comment]
I installed the mpv.app to Applications, as normal, then launched it from finder via right click - open and then giving permission in the dialog. 

Then, I added a symlink from /usr/local/bin to the binary in the app bundle like this:
`ln -s /Applications/mpv.app/Contents/MacOS/mpv /usr/local/bin/mpv`

**Comment by Doublefire-Chen** [comment]
> Hi, not really an 'idea' but just a tip when it comes to using Feishin on macOS.
> 
> The web player works out of the box, but in order to use mpv you will have to do the following:
> 
> 1. Remove the .app extension from the mpv application, turning it into a browsable folder
> 2. In Settings > Playback, set MPV Executable Path to the mpv Unix Executable (Contents > MacOS > mpv)
> 
> This wasn't immediately clear to me, so maybe should be included somewhere in the readme?

Wow, it is amazing! It works for me. Thank you, bro.
I have tried ```brew install mpv```, but it failed. Your method is smart and awesome!

**Comment by francisprovost** [comment]
For those stumbling on this, here's what work for me (MacOS 26)

- Download MPV from the link in the readme
- Extract and move the `mpv.app` to the Applications folder
- Tried to run the app once (do NOT accept the prompt to move to trash) (this step might not be necessary)
- ran `xattr -r -d com.apple.quarantine /Applications/mpv.app` to remove from quarantine (I hot a permission denied on one file though...)
- Ran the app one more time and it opened
- Removed the `.app` extension from the app like OP suggested
- Set mpv excutable in the settings to `/Applications/mpv/Contents/MacOS/mpv`

Those steps worked for me. Some of them might not be necessary. You might also try `xattr -r -d com.apple.quarantine /Applications/mpv/Contents/MacOS/mpv` to remove the executable from quarantine
